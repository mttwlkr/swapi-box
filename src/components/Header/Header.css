.header {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  height: 100px;
  flex-direction: row;
}

.favorites-button-div {
  background-color: #E5BC41;
  height: 30%;
  width: 13.5%;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-left: 25px;
}

.favorites-button {
  background: #e6bc42;
  border: none;
}

img {
  padding: 5%;
/*  background-image: url(swapi-box-logo.png);*/
/*  font-family: 'Carrois Gothic', sans-serif;
  font-size: 18px;*/
/*  font-family: 'Pathway Gothic One', sans-serif;*/
/*  color: #E5BC41;*/
  margin-left: 2%;
}


