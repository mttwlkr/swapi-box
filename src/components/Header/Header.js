import React from 'react';
import './Header.css';
import logo from './swapi-box-logo.png';
import PropTypes from 'prop-types';

const Header = (props) => {
  return (
    <header className='header'>
      <img src={logo} alt="Swapi Box" />
      <div className='favorites-button-div'>
        <button
          className='favorites-button'
          name='favorites-nav'
          onClick={props.controlFunction}
        >Your Favorites
        </button>
        <span>{props.value.length}</span>
      </div>
    </header>
  );
};

Header.propTypes = {
  value: PropTypes.array,
  controlFunction: PropTypes.func
};

export default Header;