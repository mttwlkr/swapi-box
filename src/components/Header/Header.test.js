import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header.js';

describe('Header', () => {

  let wrapper, controlFunction;

  beforeEach( () => {
    controlFunction = jest.fn();
    wrapper = shallow(<Header 
      controlFunction={controlFunction} 
      value={[1, 2, 3]}/>
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});