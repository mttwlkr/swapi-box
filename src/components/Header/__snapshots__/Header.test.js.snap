// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    controlFunction={[Function]}
    value={
        Array [
            1,
            2,
            3,
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Swapi Box"
          src="swapi-box-logo.png"
/>,
        <div
          className="favorites-button-div"
>
          <button
                    className="favorites-button"
                    name="favorites-nav"
                    onClick={[Function]}
          >
                    Your Favorites
          </button>
          <span>
                    3
          </span>
</div>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Swapi Box",
          "src": "swapi-box-logo.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="favorites-button"
              name="favorites-nav"
              onClick={[Function]}
>
              Your Favorites
</button>,
            <span>
              3
</span>,
          ],
          "className": "favorites-button-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Your Favorites",
              "className": "favorites-button",
              "name": "favorites-nav",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Your Favorites",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 3,
            },
            "ref": null,
            "rendered": 3,
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Swapi Box"
            src="swapi-box-logo.png"
/>,
          <div
            className="favorites-button-div"
>
            <button
                        className="favorites-button"
                        name="favorites-nav"
                        onClick={[Function]}
            >
                        Your Favorites
            </button>
            <span>
                        3
            </span>
</div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Swapi Box",
            "src": "swapi-box-logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="favorites-button"
                name="favorites-nav"
                onClick={[Function]}
>
                Your Favorites
</button>,
              <span>
                3
</span>,
            ],
            "className": "favorites-button-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Your Favorites",
                "className": "favorites-button",
                "name": "favorites-nav",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Your Favorites",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 3,
              },
              "ref": null,
              "rendered": 3,
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
